sync/syncer.go

```diff
@@ -291,149 +399,90 @@ func (s *Syncer) Txs(mb *types.Block) ([]*types.SerializableTransaction, error)
 
 	//map and sort txs
 	if len(missing) > 0 {
-		for out := range s.fetchWithFactory(TxReqFactory(missinDB), 1) {
+		for out := range s.fetchWithFactory(NewNeighborhoodWorker(s, 1, TxReqFactory(missinDB))) {
 			ntxs := out.([]types.SerializableTransaction)
 			for _, tx := range ntxs {
-				txMap[types.GetTransactionId(&tx)] = &tx
+				if valid, err := s.ValidateTx(&tx); !valid || err != nil {
+					id := types.GetTransactionId(&tx)
+					s.Warning("tx %v not valid %v", hex.EncodeToString(id[:]), err)
```

Comment:
shorten id strings in log print

Reply:
i print the whole so people dont miss log lines like with the shortid
maybe we should add a shortid method at some point

Reply:
please shorten the string, full ids are unnecessary and hard to read